routes/favicon.coffee

```diff
@@ -0,0 +1,28 @@
+request = require 'request'
+fs = require 'fs'
+cachefile = "favicon.ico"
+
+favicon = (req, res, next) ->
+  try
+    res.send fs.readFileSync cachefile
+  catch error
+    request "https://vine.co/", (error, response, body) ->
+      return next error if error?
+      unless response.statusCode is 200
+        error = new Error "Unexpected response code #{response.statusCode} from while getting vine.co homepage"
+        error.status = 500
+        return next error
+      request (/https:\/\/[^"]*?favicon\.ico/.exec body)[0], {encoding: 'binary'}, (error, response, body) ->
+        return next error if error?
+        unless response.statusCode is 200
+          error = new Error "Unexpected response code #{response.statusCode} from while getting vine.co homepage"
```

Comment:
I corrected the error messages. Having a dedicated error makes it easier to distinguish both cases, but I can change them to pass through instead, if you prefer that.